frontend/components/wallet/WalletProvider.tsx

```diff
@@ -53,32 +54,70 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
     }
   }, [account]);
 
+  /**
+   * Disconnects the wallet, removes the account address from state and localStorage.
+   */
+  const disconnect = () => {
+    // Clear the account address from state.
+    setAccount(null);
+    // reset the graveVault address
+    setGraveVault(undefined);
+    // Remove the stored account from localStorage.
+    localStorage.removeItem('connectedAccount');
+    // If additional logic is needed for disconnecting, it should be added here.
+  };
+
   /**
    * Connects to the wallet and sets the account address in state and localStorage.
    */
   const connect = async () => {
     if (typeof window !== 'undefined' && window.lukso) {
       // Initialize a new Web3 instance using the LUKSO provider.
       const web3 = new Web3(window.lukso);
-
+      let accounts: string[] = [];
       try {
         // Reset the graveVault address when connecting
         setGraveVault(undefined);
         // Request accounts from the wallet.
-        const accounts = await web3.eth.requestAccounts();
+        accounts = await web3.eth.requestAccounts();
         console.log('Connected with', accounts[0]);
         // Update state and localStorage with the first account address.
         setAccount(accounts[0]);
```

Comment:
I know we didn't handle it previously but perhaps something we can do in another PR is check if accounts array is empty? I guess that's a scenario where there is an extension but not accounts setup yet